src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -118,20 +107,40 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             }
         }
 
-        private async Task<AuthenticationResult> FetchNewAccessTokenAsync(CancellationToken cancellationToken)
+        private async Task<AuthenticationResult> FetchNewAccessTokenAsync(CancellationToken cancellationToken, bool proactivelyRefreshed = false)
```

Comment:
Consider not providing a default for `proactivelyRefreshed`. Makes the calling code easier to read.